File: backend/domain/user/entity.go
package user

import (
	"time"
)

type User struct {
	UserId    string
	UserName  string
	Email     Email
	CreatedAt time.Time
	UpdatedAt time.Time
}

func NewUser(userId, userName string, email Email) User {
	createdAt := time.Now()
	updatedAt := time.Now()

	return User{
		UserId:    userId,
		UserName:  userName,
		Email:     email,
		CreatedAt: createdAt,
		UpdatedAt: updatedAt,
	}
}

File: frontend/src/domain/user/entity.ts
import { z } from "zod";
import { type Email, emailSchema } from "./value-object";

export type User = {
	userId: string;
	userName: string;
	email: Email;
	createdAt: Date;
	updatedAt: Date;
};

export const userSchema: z.ZodType<User> = z.object({
	userId: z.string(),
	userName: z.string(),
	email: emailSchema,
	createdAt: z.date(),
	updatedAt: z.date(),
});
